### PR TITLE
feat: add evicted state and restake possibility

### PR DESCRIPTION
**Pull request by Tanya-atatakai**
1) Moves columns as per updated design
2) Adds epoch length column
3) Changes olas staked to staking status - can be not staked, staked * staked OLAS or evicted
4) if evicted:
 - if can restake, displays relevant button which calls unstake and then stake to the same contract
 - if can't restake due to being staked not enough time (used the logic from Pearl app) - the button is disabled and the message is shown saying when a user can restake
5) if evicted or unstaked, the current num of tweets made this epoch is alway 0

### Evicted state
Screenshot 2024-11-15 at 18 40 53

### Evicted state tooltip
Screenshot 2024-11-15 at 18 41 07

### Restake disabled message
Screenshot 2024-11-15 at 18 40 58

### CODE REVIEW

File: common-util/Login/LoginV2.jsx [hunk 1]
```diff
@@ -102,8 +102,8 @@ export const LoginV2 = ({
           if (modalProvider?.on) {
             // https://docs.ethers.io/v5/concepts/best-practices/#best-practices--network-changes
             const handleChainChanged = () => {
-              // Temp hack not to reload page when switch to base chain on staking page
-              if (!window.location.pathname.includes('staking')) {
+              // Temp hack not to reload page when switch to base chain on staking or profile page
+              if (!['staking', 'profile'].some(segment => window.location.pathname.includes(segment))) {
```
Comment (Tanya-atatakai) [inline common-util/Login/LoginV2.jsx:106]: need to better handle this next time, especially the more buttons we unlock on profile page...

File: components/Profile/Staking.jsx [hunk 2]
```diff
@@ -111,64 +122,135 @@ const StakingDetails = ({ profile }) => {
...
+    setIsRestaking(true);
+
+    try {
+      // Switch to base
+      if (chainId !== base.id) {
+        await switchChainAsync({ chainId: base.id });
+      }
+
+      // First unstake
+      await unstake({ account })
```
Comment (Tanya-atatakai) [inline components/Profile/Staking.jsx:139]: @kupermind you can review these two calls, restake is unstaking and staking to the same contract as used before

File: components/Profile/requests.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { base } from 'wagmi/chains';
+import { waitForTransactionReceipt } from '@wagmi/core';
+import { getContributeManagerContract } from 'common-util/Contracts';
+import { getEstimatedGasLimit } from 'common-util/functions/requests';
+import { wagmiConfig } from 'common-util/Login/config';
+
+export const unstake = async ({ account }) => {
```
Comment (Tanya-atatakai) [inline components/Profile/requests.js:7]: @kupermind here are the actual writings for unstake and stake

File: components/Profile/requests.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+export const stake = async ({ account, socialId, serviceId, stakingInstance }) => {
+  try {
+    const contract = getContributeManagerContract();
+    const stakeFn = contract.methods.stake(socialId, serviceId, stakingInstance);
+    const estimatedGas = await getEstimatedGasLimit(stakeFn, account);
+    const result = await stakeFn
+      .send({ from: account, gas: estimatedGas });
+    const receipt = await waitForTransactionReceipt(wagmiConfig, { chainId: base.id, hash: result.transactionHash });
```
Comment (truemiller) [inline components/Profile/requests.js:29]: 👍

**Review by kupermind — APPROVED** [review]

**Pull request merged**